### ISSUE TITLE
Process for keeping translations up to date

### ISSUE THREAD

**Issue by katrinleinweber** [github/opensource.guide#1119]
How about we collect some advice in [docs/translations.md](https://github.com/github/opensource.guide/blob/7b5aebc02a71b1f9f62873757a0643c0668612bd/docs/translations.md) under `Keeping translations up to date` or similar?

Working on #1118 I naively scrolled through the [commit list](https://github.com/github/opensource.guide/commits/master) and looked at merge commits that seemed to add content. Would `git diff $translation_branch master _articles/*.md` or similar be better?

**Comment by MikeMcQuaid** [comment]
This seems like a good idea. Could you submit a PR with your recommended workflow?

**Comment by katrinleinweber** [comment]
The snake seems to be eating its tail: [#1121/files#diff](https://github.com/github/opensource.guide/pull/1121/files#diff-31669052b00291ed5f9e4cd6db9fdd6bR31-R33) ;-)

I'll revisit this in a few days once there is a difference between my `translation_branch`& `upstream/master` to check whether the above initial idea is viable.

**Comment by syaifudin85** [comment]
join

**Comment by katrinleinweber** [comment]
This is a tough nut. I tried the following today:

1. Copy ID of last own commit in [_articles/de](https://github.com/github/opensource.guide/commits/master/_articles/de).
1. `git switch master && git pull upstream/master`
1. `git diff --word-diff $LAST_OWN upstream/master _articles/*.md > TODO-translate.txt`
1. `git switch $translation_branch`
1. `git rebase master`
1. Go through `TODO-translate.txt` & implement the English source's changes accordingly in the `_articles/**/*.md` translation(s). Because of the "own commit" step initially, one will sometimes notice already implemented changes by other translators. Skipping those seems less problematic, than remembering to only update the `translation_branch` _after_ `diff`ing.

**Comment by webknjaz** [comment]
@katrinleinweber FWIW this is something I suggested a while back: https://github.com/github/opensource.guide/issues/288#issuecomment-283927635. It'd be nice to have it fully scripted so that it'd be easier to automatically check it...

**Comment by katrinleinweber** [comment]
Yeah, that might be a more convenient source getting an ID for `3.`, but it would require each translation session to cover all of the English updates, wouldn't it?

Because those can occur in parallel to minor translation updates (correcting links, wording, or spelling), there is a risk of overlooking significant English updates that occured before such a minor update's SHA is recorded in the translation metadata, no?

**Comment by popcorner** [comment]
What about considering using some professional translate tools instead of using Git directly?
Git is powerful but not the best tool for translating articles. Contributors have to use additional steps to check the difference, and it seems hard to automatically merge new contents into translation. And, The smallest unit of Git comparison is the lines, not sentences. These problems mean additional manual work.
But these professional translate tools like Transifex, Crowdin, and Weblate can do these things automatically, with lots of additional features. They can cut articles into sentences and track all the modifications of source articles. They can handle all the issues about translation from minor changes to new articles. Translators only need to translate all the sentences that marked as "need to be translated", and all the other things like generating the translation files can be done by the platform. And these tools can also help check the translation itself, like machine translation hints, word lists to keep the same word to have the same translation, and similarity check to help translators see the existing translation from these similar sentences and avoid duplicate work. These platforms can work with Git and commit changes, and all the other workflow won't be affected. These tools provide free services to open source projects, so maybe it's a good idea to consider using these platforms to make the translations easier to maintain.

**Comment by MikeMcQuaid** [comment]
> What about considering using some professional translate tools instead of using Git directly?

This is being discussed in https://github.com/github/opensource.guide/issues/1133 and is on my list to investigate further (hopefully over Christmas).

**Comment by popcorner** [comment]
@MikeMcQuaid That tool seems to be different from the platforms I suggested. GitLocalize looks like a lightweight translation tool working with Git, and what I suggested are professional translation platforms with lots of additional features. And these platforms are more famous (according to the number of search results) and their documents are better. 
For example, https://github.com/github/opensourcefriday are using Transifex. Maybe you can try to visit [their project](https://www.transifex.com/github-open-source/open-source-friday/dashboard/) and see which type of tool is more suitable for this repository.

**Comment by MikeMcQuaid** [comment]
> For example, https://github.com/github/opensourcefriday are using Transifex. Maybe you can try to visit [their project](https://www.transifex.com/github-open-source/open-source-friday/dashboard/) and see which type of tool is more suitable for this repository.

I created the translation solution for both this project and Open Source Friday. The Transifex/Open Source Friday solution is unsuitable here as this repository is a Markdown document-based translation flow rather than lots of small strings.

https://github.com/github/opensource.guide/issues/1133 will be the next investigation and, if it seems an improvement on what we currently have, will be the next method of translation for this repository.

**Comment by foofaa222** [comment]
👍🏼

**Comment by 137717unity** [comment]
https://www.openarchives.org/rs/0.9.1/resourcesync

**Comment by smpbl** [comment]
GOOD

**Comment by horace1981** [comment]
Hi

**Comment by Jun6plushub** [comment]
1119

**Comment by 666339** [comment]
Hello

**Comment by bigwill1200** [comment]
> How about we collect some advice in [docs/translations.md](https://github.com/github/opensource.guide/blob/7b5aebc02a71b1f9f62873757a0643c0668612bd/docs/translations.md) under `Keeping translations up to date` or similar?
> 
> Working on #1118 I naively scrolled through the [commit list](https://github.com/github/opensource.guide/commits/master) and looked at merge commits that seemed to add content. Would `git diff $translation_branch master _articles/*.md` or similar be better?

You are right

**Comment by bigwill1200** [comment]
Yes

**Comment by TroySteven** [comment]
gi

**Comment by AngelaMay27** [comment]
> How about we collect some advice in [docs/translations.md](https://github.com/github/opensource.guide/blob/7b5aebc02a71b1f9f62873757a0643c0668612bd/docs/translations.md) under `Keeping translations up to date` or similar?
> 
> Working on #1118 I naively scrolled through the [commit list](https://github.com/github/opensource.guide/commits/master) and looked at merge commits that seemed to add content. Would `git diff $translation_branch master _articles/*.md` or similar be better?

**Comment by Jeedneux89** [comment]
Nice

**Comment by marknguyen92** [comment]
> How about we collect some advice in [docs/translations.md](https://github.com/github/opensource.guide/blob/7b5aebc02a71b1f9f62873757a0643c0668612bd/docs/translations.md) under `Keeping translations up to date` or similar?
> 
> Working on #1118 I naively scrolled through the [commit list](https://github.com/github/opensource.guide/commits/master) and looked at merge commits that seemed to add content. Would `git diff $translation_branch master _articles/*.md` or similar be better?

**Comment by jor514** [comment]
yes

**Comment by jor514** [comment]
yes

**Comment by jor514** [comment]
yes

**Comment by kbptomo2020** [comment]
mention: keithbrianptomo

**Comment by spiritualconnection3** [comment]
The process for either has its own individual guidelines and or application
techniques that must be applied correctly to the the array as well as root
depending on if the code is consistantly progressing in transition vs
evelutionary progress used in patching older to newer.  In the event that
isn't possible, it may be that the solution may be more complicated than
needs to be. Try a simple fork patch. Basically apply a fork at the
transition. Let me know what you think and /or if it solves the glitch.
Hope that helpselps.

**Comment by hussenasseb** [comment]
Stop

**Comment by devdev606** [comment]
Hello world

**Comment by ding243** [comment]
Hello world

**Comment by toomoon1979** [comment]
Can you help me to watch iptv for word Chanel. If any one can get me how can download iptv m3u link Iam very thankful for get me that . 
Regarding your

**Comment by Paola954** [comment]
https://github.com/github/opensource.guide/issues/1119#issuecomment-697133532

**Comment by ChenChunMing1986** [comment]
> How about we collect some advice in [docs/translations.md](https://github.com/github/opensource.guide/blob/7b5aebc02a71b1f9f62873757a0643c0668612bd/docs/translations.md) under `Keeping translations up to date` or similar?
> 
> Working on #1118 I naively scrolled through the [commit list](https://github.com/github/opensource.guide/commits/master) and looked at merge commits that seemed to add content. Would `git diff $translation_branch master _articles/*.md` or similar be better?

**Comment by LeXTeR02** [comment]
Verzonden vanaf mijn Huawei-telefoon-------- Oorspronkelijk bericht --------Van: toomoon1979 <notifications@github.com>Datum: wo 23 sep. 2020 06:56Aan: "github/opensource.guide" <opensource.guide@noreply.github.com>Cc: Subscribed <subscribed@noreply.github.com>Onderwerp: Re: [github/opensource.guide] Process for keeping translations up to date (#1119)
Can you help me to watch iptv for word Chanel. If any one can get me how can download iptv m3u link Iam very thankful for get me that .
Regarding yourDownload '' 'KODI' '' and install it then set upbit yhats need some thricks and knowledge,,, You cafind something like that on YouTube video's and try some channels,, i can promise you if its set-up correctly then ya have the best tv and cinema app of our round boll. Try some channels on YouTube, video's till you fond the set-up correct. If it wont work i know more apps for you.. contact me back. But try it,, its the best.  On KODI, really you have everything from past till far in the future REALLY. try and deffently METYou are receivingME KNOW this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by mmoo301** [comment]
#1741 #1119

**Comment by Alibhutta24** [comment]
> This seems like a good idea. Could you submit a PR with your recommended workflow?

Quite

**Comment by Alibhutta24** [comment]
> #1741 #1119

Quite

**Comment by Alibhutta24** [comment]
> [#1119 (comment)](https://github.com/github/opensource.guide/issues/1119#issuecomment-697133532)

Quite

**Comment by Alibhutta24** [comment]
> Can you help me to watch iptv for word Chanel. If any one can get me how can download iptv m3u link Iam very thankful for get me that .
> Regarding your

Quite

**Comment by Alibhutta24** [comment]
> Hello world

Quite

**Comment by Alibhutta24** [comment]
> > How about we collect some advice in [docs/translations.md](https://github.com/github/opensource.guide/blob/7b5aebc02a71b1f9f62873757a0643c0668612bd/docs/translations.md) under `Keeping translations up to date` or similar?
> > Working on #1118 I naively scrolled through the [commit list](https://github.com/github/opensource.guide/commits/master) and looked at merge commits that seemed to add content. Would `git diff $translation_branch master _articles/*.md` or similar be better?

Quiye

**Comment by Alibhutta24** [comment]
Quite

**Comment by Alibhutta24** [comment]
Close

**Comment by Alibhutta24** [comment]
> How about we collect some advice in [docs/translations.md](https://github.com/github/opensource.guide/blob/7b5aebc02a71b1f9f62873757a0643c0668612bd/docs/translations.md) under `Keeping translations up to date` or similar?
> 
> Working on #1118 I naively scrolled through the [commit list](https://github.com/github/opensource.guide/commits/master) and looked at merge commits that seemed to add content. Would `git diff $translation_branch master _articles/*.md` or similar be better?

QUITE

**Comment by LoolliH** [comment]
#2073

**Comment by LoolliH** [comment]
هونكري نوديو

**Comment by LoolliH** [comment]
تومر هيرا

**Comment by LoolliH** [comment]
توربوني سودي

**Comment by LoolliH** [comment]
اوي ناي

**Comment by LoolliH** [comment]
انكاكوريرلي

**Comment by LoolliH** [comment]
لي احد  كبير

**Comment by LoolliH** [comment]
قريب

**Comment by abdullahkhan7127165** [comment]
Duplicate of #

**Comment by MikeMcQuaid** [comment]
Closing this out. I agree a process for this would be ideal but I don't think there's anything that we (GitHub) are planning on doing here in the near future. Would welcome a pull request for this, though.
